spec/requests/conversion_hosts_spec.rb

```diff
@@ -45,4 +45,203 @@
+      expect(results.first["resource_id"]).to eq(vm.id.to_s)
+      expect(results.last["resource_id"]).to eq(host.id.to_s)
+    end
+  end
+
+  context "delete" do
+    let(:conversion_host)             { FactoryGirl.create(:conversion_host) }
+    let(:conversion_host_url)         { api_conversion_host_url(nil, conversion_host) }
+    let(:invalid_conversion_host_url) { api_conversion_host_url(nil, 999_999) }
+
+    it "can delete a conversion host via DELETE" do
+      api_basic_authorize(action_identifier(:conversion_hosts, :delete))
+      delete(conversion_host_url)
+
+      expect(response).to have_http_status(:no_content)
+      expect(ConversionHost.exists?(conversion_host.id)).to be_falsey
+    end
+
+    it "can delete a conversion host via POST" do
+      api_basic_authorize(action_identifier(:conversion_hosts, :delete, :resource_actions))
+      post(conversion_host_url, :params => gen_request(:delete))
+
+      expect_single_action_result(:success => true, :message => "deleting", :href => conversion_host_url)
+      expect(ConversionHost.exists?(conversion_host.id)).to be_falsey
+    end
+
+    it "will not delete a conversion host unless authorized" do
+      api_basic_authorize
+      delete(conversion_host_url)
+
+      expect(response).to have_http_status(:forbidden)
+      expect(ConversionHost.exists?(conversion_host.id)).to be_truthy
+    end
+  end
+
+  context "tags" do
+    let(:tag1) { {:category => "department", :name => "finance", :path => "/managed/department/finance"} }
+    let(:tag2) { {:category => "cc",         :name => "001",     :path => "/managed/cc/001"} }
+
+    let(:invalid_tag_url) { api_tag_url(nil, 999_999) }
+
+    let(:conversion_host) { FactoryGirl.create(:conversion_host) }
+
+    before do
+      FactoryGirl.create(:classification_department_with_tags)
+      FactoryGirl.create(:classification_cost_center_with_tags)
+    end
+
+    it "can list the tags for a conversion host" do
+      Classification.classify(conversion_host, tag1[:category], tag1[:name])
+      api_basic_authorize
+      get(api_conversion_host_tags_url(nil, conversion_host))
+
+      expect(response.parsed_body).to include("subcount" => 1)
+      expect(response).to have_http_status(:ok)
+    end
+
+    it "can assign a tag to a conversion host" do
+      api_basic_authorize(subcollection_action_identifier(:conversion_hosts, :tags, :assign))
+
+      post(api_conversion_host_tags_url(nil, conversion_host), :params => { :action => "assign", :category => "department", :name => "finance" })
+
+      expected = {
+        "results" => [
+          a_hash_including(
+            "success"      => true,
+            "message"      => a_string_matching(/assigning tag/i),
+            "tag_category" => "department",
+            "tag_name"     => "finance"
+          )
+        ]
+      }
+      expect(response.parsed_body).to include(expected)
+      expect(response).to have_http_status(:ok)
```

Comment:
Do we want to verify the tag is assigned?

Reply:
Ok, done.